### PR TITLE
Cleanup Brancher nodes by labels

### PR DESCRIPTION
**Pull request by tdgroot**
Depends on and needs to be merged after #65.

This allows brancher nodes to be cleaned up by labels. Hypernode Deploy will look up existing brancher nodes using the API, match them with the labels of the given stage and delete them.

This is an explicit feature, so it is necessary to call `hypernode-deploy cleanup <target>` on a specific point of the pipeline configuration (for example after a PR is closed).

### CODE REVIEW

**Review by AlexanderGrooff — COMMENTED** [review]
I see you're changing some of the configuration loading. Is that necessary for specifying extra input for labels?

File: src/DeployRunner.php [hunk 1]
```diff
@@ -427,4 +343,19 @@ public function getDeploymentReport()
...
+    {
+        $console = new Application();
+        $deployer = new Deployer($console);
+        $deployer['output'] = new OutputWatcher($output);
+        $deployer['input'] = new ArrayInput(
+            [],
+            new InputDefinition([
+                new InputOption('limit'),
+                new InputOption('profile'),
+            ])
```
[the remark this answers is not in the document]
Comment (tdgroot) [reply src/DeployRunner.php:357]: This was initially done in `DeployRunner::run`, but extracted it into a separate method to clean things up

File: src/DeployRunner.php [hunk 2]
```diff
@@ -360,65 +335,6 @@ private function runStage(Deployer $deployer, string $stage, string $task = 'dep
         return $exitCode;
     }
 
-    private function tryGetConfiguration(): Configuration
```
[the remark this answers is not in the document]
Comment (tdgroot) [reply src/DeployRunner.php:363]: I moved configuration loading into a separate unit, namely `ConfigurationLoader`. This way we can load a configuration file outside of the `DeployRunner` (like we need in `Cleanup`). Also separation of concerns :).

File: src/Brancher/BrancherHypernodeManager.php [hunk 3]
```diff
@@ -22,6 +22,36 @@ public function __construct(LoggerInterface $log)
...
+     *
+     * @param string $hypernode The parent hypernode to query the Brancher instances from
+     * @param string[] $labels Labels to match against, may be empty
+     * @return string[] The found Brancher instance names
+     */
+    public function queryBrancherHypernodes(string $hypernode, array $labels = []): array
+    {
+        $result = [];
+
+        $hypernodes = $this->hypernodeClient->app->getList(['parent' => $hypernode, 'type' => 'brancher']);
```
[the remark this answers is not in the document]
Comment (tdgroot) [reply src/Brancher/BrancherHypernodeManager.php:36]: Yeah I think that would be the right approach, but let's do that in a later stage

**Comment by tdgroot** [comment]
> I see you're changing some of the configuration loading. Is that necessary for specifying extra input for labels?

The configuration loading was moved into a separate class and I also removed the `if fail then do composer install and retry`, because it's really unclear what use case we're covering there. So it's way simpler now and also reusable in other components.

**Review by vdloo — APPROVED** [review]

**Pull request merged**